File: kirun-js/__tests__/engine/function/system/string/DeleteForGivenLengthTest.ts
import { KIRunFunctionRepository, KIRunSchemaRepository } from '../../../../../src';
import { DeleteForGivenLength } from '../../../../../src/engine/function/system/string/DeleteForGivenLength';
import { SchemaValidationException } from '../../../../../src/engine/json/schema/validator/exception/SchemaValidationException';
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';
import { MapUtil } from '../../../../../src/engine/util/MapUtil';

const deleteT: DeleteForGivenLength = new DeleteForGivenLength();

test('DeleteForGivenLength test1', async () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    );

    fep.setArguments(
        new Map<string, string | number>([
            [DeleteForGivenLength.PARAMETER_STRING_NAME, ' THIScompatY IS A NOcoDE plATFNORM'],
            [DeleteForGivenLength.PARAMETER_AT_START_NAME, 10],
            [DeleteForGivenLength.PARAMETER_AT_END_NAME, 18],
        ]),
    );

    let outputString: string = ' THIScompaNOcoDE plATFNORM';

    expect((await deleteT.execute(fep)).allResults()[0].getResult().get('result')).toBe(
        outputString,
    );
});

test('DeleteForGivenLength test2', async () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    );

    fep.setArguments(
        new Map<string, string | number>([
            [DeleteForGivenLength.PARAMETER_STRING_NAME, ' THIScompatY IS A NOcoDE plATFNORM'],
            [DeleteForGivenLength.PARAMETER_AT_START_NAME, 4],
            [DeleteForGivenLength.PARAMETER_AT_END_NAME, 10],
        ]),
    );

    let outputString: string = ' THItY IS A NOcoDE plATFNORM';

    expect((await deleteT.execute(fep)).allResults()[0].getResult().get('result')).toBe(
        outputString,
    );
});

File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/string/DeleteForGivenLengthTest.java
package com.fincity.nocode.kirun.engine.function.system.string;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

class DeleteForGivenLengthTest {

	@Test
	void test() {
		String s1 = " THIScompatY IS A NOcoDE plATFNORM";

		DeleteForGivenLength delete = new DeleteForGivenLength();

		StepVerifier.create(delete
				.execute(new ReactiveFunctionExecutionParameters(new KIRunReactiveFunctionRepository(),
						new KIRunReactiveSchemaRepository())
						.setArguments(Map.of(DeleteForGivenLength.PARAMETER_STRING_NAME, new JsonPrimitive(s1),
								DeleteForGivenLength.PARAMETER_AT_START_NAME, new JsonPrimitive(10),
								DeleteForGivenLength.PARAMETER_AT_END_NAME, new JsonPrimitive(18))))
				.map(fo -> fo.allResults().get(0).getResult().get(DeleteForGivenLength.EVENT_RESULT_NAME)
						.getAsString()))
				.expectNext(" THIScompaNOcoDE plATFNORM")
				.verifyComplete();
	}

	@Test
	void test2() {

		String s1 = " THIScompatY IS A NOcoDE plATFNORM";

		DeleteForGivenLength insert = new DeleteForGivenLength();

		StepVerifier.create(insert
				.execute(new ReactiveFunctionExecutionParameters(new KIRunReactiveFunctionRepository(),
						new KIRunReactiveSchemaRepository())
						.setArguments(Map.of(DeleteForGivenLength.PARAMETER_STRING_NAME, new JsonPrimitive(s1),
								DeleteForGivenLength.PARAMETER_AT_START_NAME, new JsonPrimitive(4),
								DeleteForGivenLength.PARAMETER_AT_END_NAME, new JsonPrimitive(10))))
				.map(fo -> fo.allResults().get(0).getResult().get(DeleteForGivenLength.EVENT_RESULT_NAME)
						.getAsString()))
				.expectNext(" THItY IS A NOcoDE plATFNORM")
				.verifyComplete();
	}
}
